vnext/Desktop/WinHTTPWebSocketResource.cpp

```diff
@@ -0,0 +1,447 @@
+    NULL,               // pwszReferrer
+    NULL,               // ppwszAcceptTypes
+    0                   // dwFlags
+  );
+
+  if (m_requestHandle == NULL)
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error starting request", ErrorType::Connection });
+    }
+  }
+
+  if (!WinHttpSetOption(m_requestHandle, WINHTTP_OPTION_CONTEXT_VALUE, this, sizeof(this)))
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error configuring request", ErrorType::Connection });
+    }
+  }
+
+  // Set WinHTTP callbacks
+  if (WinHttpSetStatusCallback(
+    m_requestHandle,                          // hInternet
+    Callback,                                 // lpfnInternetCallback
+    WINHTTP_CALLBACK_FLAG_ALL_NOTIFICATIONS,  // dwNotificationFlags
+    0                                         // dwReserved
+  ) == WINHTTP_INVALID_STATUS_CALLBACK)
+  {
+    m_errorStatus = GetLastError();
+    throw std::exception("Can not create instance. Invalid callback.");
+  }
+
+  BOOL fStatus = FALSE; //TODO: Make member?
+  fStatus = WinHttpSetOption(
+    m_requestHandle,                      // hInternet
+    WINHTTP_OPTION_UPGRADE_TO_WEB_SOCKET, // dwOption
+    NULL,                                 // lpBuffer
+    0                                     // dwBufferLength
+  );
+
+  if (!fStatus)
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error upgrading protocol", ErrorType::Connection });
+    }
+  }
+
+  // Perform handshake
+  fStatus = WinHttpSendRequest(
+    m_requestHandle,                // hRequest
+    WINHTTP_NO_ADDITIONAL_HEADERS,  // lpszHeaders
+    0,                              // dwHeadersLength
+    NULL,                           // lpOptional
+    0,                              // dwOptionalLength
+    0,                              // dwTotalLength
+    (DWORD_PTR)this                 // dwContext
+  );
+  if (!fStatus)
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error sending handshake", ErrorType::Handshake });
+    }
+  }
+  fStatus = WinHttpReceiveResponse(m_requestHandle, /*lpReserved*/ 0);
+  if (!fStatus)
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error receiving handshake", ErrorType::Handshake });
+    }
+  }
+
+  // Complete upgrade. Accept only HTTP status code 101.
+  m_webSocketHandle = WinHttpWebSocketCompleteUpgrade(m_requestHandle, /*pContext*/ NULL);
+  if (m_webSocketHandle == NULL)
+  {
+    m_errorStatus = GetLastError();
+
+    if (m_errorHandler)
+    {
+      return m_errorHandler({ "Error completing protocol upgrade", ErrorType::Connection });
+    }
+  }
+
+  // Connection succeeded.
+  if (m_connectHandler)
+  {
+    m_connectHandler();
+  }
+}
+
+void WinHTTPWebSocketResource::Ping() {}
+
```

Comment:
not implemented - do we need to have the function at all? if so, let's 
a) add debug spew that this isn't implemented
b) file an Issue to add this functionality

Reply:
I'll file an overall follow-up issue and self-asign.